server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestExecBlockWithoutEndExec.java

```diff
@@ -54,9 +54,15 @@ void testEndExecBlockRequiredForExecSQL() {
             "1",
             new Diagnostic(
                 new Range(),
-                "Missing token END-EXEC for the EXEC block",
+                "Missing token END-EXEC at execRule",
```

Comment:
I suggest adjusting the recovery strategy when error occurs in the `sqlCode` rule: If the end-exec is not available, I'd suggest rewinding back to the next dot, semi-colon (since it is supported), the next paragraph/section boundary (something starting in area a), or start of a program (identification division), whatever comes first after the `EXEC SQL`.
And actually issue the original message `Missing token END-EXEC for the EXEC block`.
It should be quite easy to do this by just examining the already available tokens.